pkg/server/sotw/v3/server.go

```diff
@@ -102,6 +102,11 @@ type watches struct {
 	nonces        map[string]string
 }
 
+type latestDiscoveryResponse struct {
```

Comment:
I think we could move things like this into the `StreamState`. That way we don't need to declare extra things just for SOTW

Reply:
A comment here noting why we need this might be helpful